### PR TITLE
curtail: init at 1.3.0

### PR DESCRIPTION
**Pull request by schuelermine**
###### Description of changes

Added Curtail, a simple GTK image compression utility, as curtail

This is a reopening of PR #173397 due to force pushes. I have tested functionality and localization.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/curtail/default.nix [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  strictDeps = false;
+
+  # Tells the Python builder to use normal building methods instead of using a setup.py
+  format = "other";
+
+  preInstall = ''
+    patchShebangs --build /build/source/build-aux/meson/postinstall.py
+  '';
+  postInstall = ''
+    wrapProgram $out/bin/curtail --prefix PATH : ${lib.makeBinPath runtimeDeps}
```
Comment (muscaln) [inline pkgs/applications/graphics/curtail/default.nix:67]:
```suggestion
    wrapProgram $out/bin/curtail --prefix PATH : ${lib.makeBinPath [ jpegoptim libwebp optipng pngquant ]}
```

File: pkgs/applications/graphics/curtail/default.nix [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+let
+  runtimeDeps = [
+    jpegoptim
+    libwebp
+    optipng
+    pngquant
+  ];
+
+in python3.pkgs.buildPythonApplication rec {
+
```
Comment (muscaln) [inline pkgs/applications/graphics/curtail/default.nix:30]:
```suggestion

```

File: pkgs/applications/graphics/curtail/default.nix [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+, meson
+, ninja
+, python3
+, gettext
+, gtk3
+, optipng
+, pngquant
+, jpegoptim
+, libwebp
+, glib
```
Comment (muscaln) [inline pkgs/applications/graphics/curtail/default.nix:12]:
```suggestion

```

**Review by muscaln — CHANGES_REQUESTED** [review]

**Comment by muscaln** [comment]
Result of `nixpkgs-review pr 173520` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>curtail</li>
  </ul>
</details>

**Comment by muscaln** [comment]
I couldn't be able to launch `curtail`. Fails with

```
❯ /nix/store/xhjk27j7skafbrwbn6l54vpxqrmqxaf7-curtail-1.3.0/bin/curtail
Traceback (most recent call last):
  File "/nix/store/zm839ag6yaapyivq3h0jzi2kc0a7fivl-python3.9-pygobject-3.42.1/lib/python3.9/site-packages/gi/importer.py", line 142, in load_module
    introspection_module = get_introspection_module(namespace)
  File "/nix/store/zm839ag6yaapyivq3h0jzi2kc0a7fivl-python3.9-pygobject-3.42.1/lib/python3.9/site-packages/gi/module.py", line 257, in get_introspection_module
    module = IntrospectionModule(namespace, version)
  File "/nix/store/zm839ag6yaapyivq3h0jzi2kc0a7fivl-python3.9-pygobject-3.42.1/lib/python3.9/site-packages/gi/module.py", line 109, in __init__
    repository.require(namespace, version)
gi.RepositoryError: Typelib file for namespace 'Pango', version '1.0' not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/nix/store/xhjk27j7skafbrwbn6l54vpxqrmqxaf7-curtail-1.3.0/bin/...curtail-wrapped-wrapped-wrapped", line 43, in <module>
    from curtail import main
  File "/nix/store/xhjk27j7skafbrwbn6l54vpxqrmqxaf7-curtail-1.3.0/share/curtail/curtail/main.py", line 23, in <module>
    from gi.repository import Gtk, Gio
  File "/nix/store/zm839ag6yaapyivq3h0jzi2kc0a7fivl-python3.9-pygobject-3.42.1/lib/python3.9/site-packages/gi/importer.py", line 144, in load_module
    raise ImportError(e)
ImportError: Typelib file for namespace 'Pango', version '1.0' not found
```

**Comment by schuelermine** [comment]
I had left out Pango in the build inputs on refactoring. I’ve added it back now.

**Comment by schuelermine** [comment]
It still doesn’t work. I seem to have messed up something.

**Comment by schuelermine** [comment]
Oh. I had actually removed GTK as a dependency! Thank you for noticing.

**Comment by muscaln** [comment]
Result of `nixpkgs-review pr 173520` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>curtail</li>
  </ul>
</details>

**Comment by muscaln** [comment]
Works fine but output has multiple wrappers.

```
❯ ls -a /nix/store/zxi8vlhvwmjrcziqdc9al9hd3ks6x88q-curtail-1.3.0/bin/
.  ..  curtail  .curtail-wrapped  .curtail-wrapped_  ..curtail-wrapped-wrapped  ...curtail-wrapped-wrapped-wrapped
```

**Comment by schuelermine** [comment]
Is that a big problem? It wraps once for the Python path, once for GApps & GIO, and once for putting the tools into the path.
I could add the tools into the extra arguments variables for one of these, but this seems more idiomatic.
Should I maybe move the two first wrappers into libexec before wrapping?

**Comment by schuelermine** [comment]
I guess they are from that. I don’t see the problem with these additional wraps.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -35371,4 +35371,6 @@ with pkgs;
   honeyvent = callPackage ../servers/tracing/honeycomb/honeyvent { };
 
   mictray = callPackage ../tools/audio/mictray { };
+
+  curtail = callPackage ../applications/graphics/curtail { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:35375]: Please sort this into a fitting category and sort there alphabetically.

File: pkgs/applications/graphics/curtail/default.nix [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    gobject-introspection
+    gettext
+  ];
+
+  # This is needed to let gobject-introspection’s setup hook run
+  # Disabling this breaks icon themes (at least)
+  strictDeps = false;
+
+  # Tells the Python builder to use normal building methods instead of using a setup.py
+  format = "other";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/curtail/default.nix:53]:
```suggestion
```

File: pkgs/applications/graphics/curtail/default.nix [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+, gobject-introspection
+, jpegoptim
+, libwebp
+, optipng
+, pngquant
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "curtail";
+  version = "1.3.0";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/curtail/default.nix:21]:
```suggestion
  version = "1.3.0";
  format = "other";
```

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (schuelermine) [reply pkgs/top-level/all-packages.nix:35375]: Done

File: pkgs/applications/graphics/curtail/default.nix (hunk 6 above)
Comment (schuelermine) [reply pkgs/applications/graphics/curtail/default.nix:21]: I don’t see why? The format declaration is already the first build-related instruction after the inputs.
Why remove the comment? Isn’t it helpful?

**Comment by schuelermine** [comment]
Sorry, I bollocksed up the branch with a wrong rebase.

**Comment by schuelermine** [comment]
I have reopened this at #174712

**Pull request closed without merging**